x-pack/plugins/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -103,16 +105,35 @@ export function ApmMainTemplate({
     status === FETCH_STATUS.LOADING ||
     fleetApmPoliciesStatus === FETCH_STATUS.LOADING;
 
+  const hasApmData = !!data?.hasData;
+  const hasApmIntegrations = !!fleetApmPoliciesData?.hasApmPolicies;
+
   const noDataConfig = getNoDataConfig({
     basePath,
     docsLink: docLinks!.links.observability.guide,
-    hasApmData: data?.hasData,
-    hasApmIntegrations: fleetApmPoliciesData?.hasApmPolicies,
+    hasApmData,
+    hasApmIntegrations,
     shouldBypassNoDataScreen,
     loading: isLoading,
     isServerless: config?.serverlessOnboarding,
   });
 
+  useEffect(() => {
+    return aiAssistant.setApplicationContext({
+      description: [
+        hasApmData
+          ? 'The user has APM data.'
+          : 'The user does not have APM data.',
+        hasApmIntegrations
+          ? 'The user has the APM integration installed. '
+          : 'The user does not have the APM integration installed',
+        !noDataConfig
+          ? ''
+          : 'The user is looking at a screen that tells them they do not have any data.',
```

Comment:
Up to you, but I like to have the truth handles first:
```suggestion
        noDataConfig
          ? 'The user is looking at a screen that tells them they do not have any data.'
          : '',
```

Reply:
Agreed, will change!